tutorials/transformers-text-classification.ipynb

```diff
@@ -0,0 +1,751 @@
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "xnLbsx2Jhurm"
+      },
+      "source": [
+        "We move towards the end of the tutorial for PyTorch specific instructions. Here we are extracting a larger subset of our original datasets. We also don't need to provide a seed since we seeded everything at the beginning."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "iPQy_FMcxBy3"
+      },
+      "source": [
+        "tokenized_datasets = tokenized_datasets.remove_columns([\"text\"])\n",
+        "tokenized_datasets = tokenized_datasets.rename_column(\"label\", \"labels\")\n",
+        "tokenized_datasets.set_format(\"torch\")\n",
+        "\n",
+        "small_train_dataset = tokenized_datasets[\"train\"].shuffle().select(range(5000))\n",
+        "small_eval_dataset = tokenized_datasets[\"test\"].shuffle().select(range(5000))"
+      ],
+      "execution_count": 6,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0JvnKjqRiEFP"
+      },
+      "source": [
+        "### Dataloaders"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "APNr5lgsygtw"
+      },
+      "source": [
+        "from torch.utils.data import DataLoader\n",
+        "\n",
+        "train_dataloader = DataLoader(small_train_dataset, shuffle=True, batch_size=8)\n",
+        "eval_dataloader = DataLoader(small_eval_dataset, batch_size=8)"
+      ],
+      "execution_count": 7,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ax8ToOAviGGS"
+      },
+      "source": [
+        "### Model"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "XVS15FELytIj"
+      },
+      "source": [
+        "from transformers import AutoModelForSequenceClassification\n",
+        "\n",
+        "model = AutoModelForSequenceClassification.from_pretrained(\"bert-base-cased\", num_labels=2)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "QOI8dLDDiI7c"
+      },
+      "source": [
+        "### Optimizer"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "RIihFsPryvI_"
+      },
+      "source": [
+        "from transformers import AdamW\n",
+        "\n",
+        "optimizer = AdamW(model.parameters(), lr=5e-5)"
+      ],
+      "execution_count": 9,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0-Yd47-UiLFs"
+      },
+      "source": [
+        "### LR Scheduler\n",
+        "\n",
+        "We will use the built-in Ignite alternative of `linear` scheduler which is [`PiecewiseLinear`](https://pytorch.org/ignite/generated/ignite.handlers.param_scheduler.PiecewiseLinear.html#piecewiselinear). We will also increase the number of epochs."
```

Comment:
Why not introduce a warmup, it's a cool reason to explain why use this scheduler instead of the PyTorch one.

Reply:
I'm trying to convert the code provided in the reference tutorial:
```f
rom transformers import get_scheduler

num_epochs = 3
num_training_steps = num_epochs * len(train_dataloader)
lr_scheduler = get_scheduler(
    "linear",
    optimizer=optimizer,
    num_warmup_steps=0,
    num_training_steps=num_training_steps
)```
and therefore I didn't introduce a warmup.

Reply:
Ok, you mentioned your intention to match to the original code, so my comment doesn’t matter.